esp/services/ws_fileio/ws_fileioservice.cpp

```diff
@@ -49,46 +49,52 @@ bool CWsFileIOEx::CheckServerAccess(const char* targetDZNameOrAddress, const cha
         Owned<IConstDropZoneInfoIterator> dropZoneItr = env->getDropZoneIteratorByAddress(targetDZNameOrAddress);
         ForEach(*dropZoneItr)
         {
-            dropZoneInfo.set(&dropZoneItr->query());
-            break;
+            IConstDropZoneInfo & dz = dropZoneItr->query();
+            if (dz.isECLWatchVisible())
+            {
+                dropZoneInfo.set(&dropZoneItr->query());
+                netAddr.set(targetDZNameOrAddress);
+                break;
+            }
         }
     }
 
     if (dropZoneInfo)
     {
-        char ch = '\\';
         SCMStringBuffer dropZoneName, directory, computerName, computerAddress;
-        dropZoneInfo->getName(dropZoneName);
-        dropZoneInfo->getComputerName(computerName); //legacy structure
-        if(computerName.length() != 0)
+        if (netAddr.isEmpty())
         {
-            Owned<IConstMachineInfo> machine = env->getMachine(computerName.str());
-            if (machine)
+            dropZoneInfo->getName(dropZoneName);
```

Comment:
This line should be moved above line 65 since the dropZoneName may be used in line 109.